### PR TITLE
feat: contract deployment

### PR DESCRIPTION
**Pull request by mathewmeconry**

### CODE REVIEW

**Review by nivida — CHANGES_REQUESTED** [review]
LGTM! But on merge into main should it only update the factory contracts and not the Registry cause there in case we have to update it are some further steps required. Also it would be great to have the new addresses of the factories added together with a commit hash to a deployments.md or similar in this repository.

**Comment by mathewmeconry** [comment]
> LGTM! But on merge into main should it only update the factory contracts and not the Registry cause there in case we have to update it are some further steps required. Also it would be great to have the new addresses of the factories added together with a commit hash to a deployments.md or similar in this repository.

done
it updates the `Releases.md` file in `packages/contracts` and updates the `active_contracts.json` file in the root of this repository.
`active_contracts.json` can then be reused for other deployments

**Review by nivida — APPROVED** [review]
LGTM! thanks! :))

File: packages/contracts/scripts/deploy.ts [hunk 1]
```diff
@@ -3,28 +3,134 @@
 //
 // When running the script with `npx hardhat run <script>` you'll find the Hardhat
 // Runtime Environment's members available in the global scope.
-import { ethers } from "hardhat";
+import {ethers} from 'hardhat';
```
Comment (Rekard0) [inline packages/contracts/scripts/deploy.ts]: i am a bit confused. why we use hardhat script when we have hardhat-deploy plugin installed and used. check package.json and /deploy

or i am missing something?

Comment (mathewmeconry) [reply packages/contracts/scripts/deploy.ts]: valid point. haven't noticed this plugin.
I'll restructure it with the refactoring to support multiple networks properly

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**